### DISCUSSION TITLE
rate limit: match multiple domains in one limit

### DISCUSSION THREAD

**Discussion by yvan-algoo** [rspamd/rspamd#4530] [Q&A] [answered]
Hi,

In the rate limit module, I am trying to match multiple domains in one limit.

I first tried with a selector and a regexp but can not find the proper syntax of the "or" part (probably because it is a Lua regexp which can not "or" but I am not sure):
```
  selector = from('smtp'):domain.regexp("^(domain1.com|domain2.com)$")
```
Then I tried with a selector and the ".in()" transformation but again can not find the proper syntax:
```
  selector = from('smtp'):domain.in("domain1.com","domain2.com")
```
Is the above possible, or must I write a Lua "user-defined ratelimit" as written in the doc (https://rspamd.com/doc/modules/ratelimit.html#user-defined-ratelimits)?

Regards,
Yvan

**Comment by fatalbanana** [comment]


**Reply by yvan-algoo** [reply]
Thanks @fatalbanana, this seems great. At the same time, I corrected the selector with regexp:
```
selector = from('smtp'):domain.regexp("^(domain1\.com|domain2\.com)$")
```
While syntactically correct, it creates two buckets in Redis (meaning one accounting per domain), while I need **one accounting for multiple domains**: don't you think that your solution will have the same effect?

**Reply by fatalbanana** [reply]
Yes, indeed. You could use `apply_map` instead of `filter_map` and set values in the map, corresponding to some name for the group of domains.

**Reply by yvan-algoo** [reply]
Thanks, I will try that when time permits and let you know.

**Comment by vstakhov** [comment, marked as answer]
`selector = from('smtp'):domain.regexp("^(domain1\.com|domain2\.com)$").id('my_bucket')`

**Reply by yvan-algoo** [reply]
Thanks @vstakhov this works perfectly!
